### DISCUSSION TITLE
Cron Job doesn't run at 00:00 in GitHub Actions

### DISCUSSION THREAD

**Discussion by ghost** [community/community#52477] [Actions]
### Select Topic Area

Question

### Body

Hi everyone, I've been trying to set up a cron job in GitHub Actions to run a workflow every day at midnight. I used the following configuration for the schedule trigger: 

```
 on: 
        schedule: 
            - cron: '0 0 * * 1-5' 
``` 
However, the job doesn't seem to be running at exactly midnight. Sometimes it runs a few minutes later, and other times it doesn't seem to run at all. I've checked the time zone of my runner and it's set correctly, so I'm not sure what the issue could be.
Has anyone else experienced this problem before? Does anyone have any suggestions for how I can get my cron job to run reliably at midnight?

I also noticed that upgrading to an enterprise account might help with this issue. Does anyone have experience with this? 

Thanks in advance for your help.

**Comment by airtower-luna** [comment]
This is [documented behavior](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) for `schedule` events:

> Note: The schedule event can be delayed during periods of high loads of GitHub Actions workflow runs. High load times include the start of every hour. If the load is sufficiently high enough, some queued jobs may be dropped. To decrease the chance of delay, schedule your workflow to run at a different time of the hour.

If you need reliable execution at an exact time, best get dedicated infrastructure instead of relying on Actions.

**Comment by woctezuma** [comment]
Hello,

I think this is what happens with one of my Action: the job is skipped even though it is scheduled at 00:35 AM.
The previous event at 23:35 PM and the next event at 01:35 AM both work fine.

So I imagine that Github is under high load during a long time starting at midnight: it is not enough to delay it by half an hour or so.

I was afraid that there would be something wrong with my schedule, shown below, as the issue happens between the two days (Wednesday and Thursday), but I believe it is a coincidence, and that the issue is the high load at midnight.

```yml
on:
  workflow_dispatch:
  schedule:
    - cron: '35 18-23 * * 3'
    - cron: '35 0-12 * * 4'
```

**Edit:** [I realize](https://github.com/orgs/community/discussions/73251) the hourly job is not skipped, it is delayed by close to an hour (hence my confusion).

![Picture](https://i.imgur.com/MaHejed.png)
